### ISSUE TITLE
iomisc_reg failure on aarch64

### ISSUE THREAD

**Issue by manisandro** [DanBloomberg/leptonica#394]
While working on updating to the Fedora leptonica package to 1.77, I hit the following test failure on aarch64 (other arches built fine [1]):

    FAIL: iomisc_reg
    ================
    ////////////////////////////////////////////////
    ////////////////   iomisc_reg   ///////////////
    ////////////////////////////////////////////////
    leptonica-1.77.0 : libgif 5.1.4 : libjpeg 6b (libjpeg-turbo 2.0.0) : libpng 1.6.35 : libtiff 4.0.10 : zlib 1.2.11 : libwebp 1.0.0
    Info in pixReadStreamPng: not stripping 16 --> 8 in png reading
    Failure in iomisc_reg: value comparison for index 6
    difference = 2875.000000 but allowed delta = 10.000000
    Failure in iomisc_reg: value comparison for index 7
    difference = 309.000000 but allowed delta = 10.000000
    Info in pixReadStreamPng: not stripping 16 --> 8 in png reading
    Info in pixReadStreamPng: not stripping 16 --> 8 in png reading
    Info in pixReadStreamPng: not stripping 16 --> 8 in png reading
    Info in pixReadStreamPng: not stripping 16 --> 8 in png reading
    Failure in iomisc_reg: value comparison for index 36
    difference = 124.000000 but allowed delta = 0.000000
    Time:   0.406 sec
    ////////////////////////////////////////////////
    ////////////////   iomisc_reg   ///////////////
    ////////////////////////////////////////////////
    leptonica-1.77.0 : libgif 5.1.4 : libjpeg 6b (libjpeg-turbo 2.0.0) : libpng 1.6.35 : libtiff 4.0.10 : zlib 1.2.11 : libwebp 1.0.0
    Info in pixReadStreamPng: not stripping 16 --> 8 in png reading
    Failure in iomisc_reg: value comparison for index 6
    difference = 2875.000000 but allowed delta = 10.000000
    Failure in iomisc_reg: value comparison for index 7
    difference = 309.000000 but allowed delta = 10.000000
    Info in pixReadStreamPng: not stripping 16 --> 8 in png reading
    Info in pixReadStreamPng: not stripping 16 --> 8 in png reading
    Info in pixReadStreamPng: not stripping 16 --> 8 in png reading
    Info in pixReadStreamPng: not stripping 16 --> 8 in png reading
    Failure in iomisc_reg: value comparison for index 36
    difference = 124.000000 but allowed delta = 0.000000
    Time:   0.425 sec
    FAIL iomisc_reg (exit status: 1)

[1] https://koji.fedoraproject.org/koji/taskinfo?taskID=31530991

**Comment by DanBloomberg** [comment]
That is surprising!  I wonder why the library gives different results on different platforms.

It's easily fixed.  In the following lines, change the last arg in the function regTestCompareValues()

  line 94        10.0 --> 3000.0
  line 271        0.  -->  200.0

Let me know if that fixes it.  If so, I'll change the test with a comment.  What is aarch64?

  -- Dan

**Comment by DanBloomberg** [comment]
also, note that in iomisc_regI have mis-numbered some of the tests.  The one on line 271 is actually #36, as described in the error output, even though I commented it as #35.

**Comment by DanBloomberg** [comment]
OK, I'll push the change in the next week or so, and close this issue.

Thank you for bringing it up!

  -- Dan

**Comment by frispete** [comment]
FWIW, the second hunk fixed my issues to build 1.77 for an older openSUSE release.

Without the patch, it broke reproducible with:
```
////////////////////////////////////////////////
////////////////   iomisc_reg   ///////////////
////////////////////////////////////////////////
leptonica-1.77.0 : libjpeg 8d (libjpeg-turbo 1.3.1) : libpng 1.6.36 : libtiff 4.0.7 : zlib 1.2.8 : libwebp 0.4.1 : libopenjp2 2.1.0
Info in pixReadStreamPng: not stripping 16 --> 8 in png reading
Info in pixReadStreamPng: not stripping 16 --> 8 in png reading
Info in pixReadStreamPng: not stripping 16 --> 8 in png reading
Info in pixReadStreamPng: not stripping 16 --> 8 in png reading
Info in pixReadStreamPng: not stripping 16 --> 8 in png reading
Failure in iomisc_reg, index 37: comparing /tmp/lept/io/fract8.tif with /tmp/lept/golden/iomisc_golden.37.tif
Failure in iomisc_reg: value comparison for index 38
difference = 12.000000 but allowed delta = 0.000000
Time:   0.250 sec
```
**Update**: obviously, this issue is related to the tiff version, after updating to libtiff 4.0.11, patch isn't needed anymore:
```
////////////////////////////////////////////////
////////////////   iomisc_reg   ///////////////
////////////////////////////////////////////////
leptonica-1.77.0 : libjpeg 8d (libjpeg-turbo 1.3.1) : libpng 1.6.36 : libtiff 4.0.10 : zlib 1.2.8 : libwebp 0.4.1 : libopenjp2 2.1.0
Info in pixReadStreamPng: not stripping 16 --> 8 in png reading
Info in pixReadStreamPng: not stripping 16 --> 8 in png reading
Info in pixReadStreamPng: not stripping 16 --> 8 in png reading
Info in pixReadStreamPng: not stripping 16 --> 8 in png reading
Info in pixReadStreamPng: not stripping 16 --> 8 in png reading
Time:   0.189 sec
```
The related build is available in public [here](https://build.opensuse.org/package/show/home:frispete:13.2/leptonica).
